jazz_cognito-authorizer/index.js

```diff
@@ -165,330 +125,370 @@ function ValidateToken(pems, event, context, cb) {
+            this.restApiId = "*";
+        } else {
+            this.restApiId = apiOptions.restApiId;
+        }
+        if (!apiOptions || !apiOptions.region) {
+            this.region = "*";
+        } else {
+            this.region = apiOptions.region;
+        }
+        if (!apiOptions || !apiOptions.stage) {
+            this.stage = "*";
+        } else {
+            this.stage = apiOptions.stage;
+        }
+    }
+
+    /**
+     * A set of existing HTTP verbs supported by API Gateway. This property is here
+     * only to avoid spelling mistakes in the policy.
+     *
+     * @property HttpVerb
+     * @type {Object}
+     */
+    AuthPolicy.HttpVerb = {
+        GET: "GET",
+        POST: "POST",
+        PUT: "PUT",
+        PATCH: "PATCH",
+        HEAD: "HEAD",
+        DELETE: "DELETE",
+        OPTIONS: "OPTIONS",
+        ALL: "*"
+    };
 
+    AuthPolicy.prototype = (function () {
         /**
-         * Adds an API Gateway method (Http verb + Resource path) to the list of allowed
-         * methods and includes a condition for the policy statement. More on AWS policy
-         * conditions here: http://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements.html#Condition
+         * Adds a method to the internal lists of allowed or denied methods. Each object in
+         * the internal list contains a resource ARN and a condition statement. The condition
+         * statement can be null.
          *
-         * @method allowMethodWithConditions
-         * @param {String} The HTTP verb for the method, this should ideally come from the
+         * @method addMethod
+         * @param {String} The effect for the policy. This can only be "Allow" or "Deny".
+         * @param {String} he HTTP verb for the method, this should ideally come from the
          *                 AuthPolicy.HttpVerb object to avoid spelling mistakes
-         * @param {string} The resource path. For example "/pets"
-         * @param {Object} The conditions object in the format specified by the AWS docs
+         * @param {String} The resource path. For example "/pets"
+         * @param {Object} The conditions object in the format specified by the AWS docs.
          * @return {void}
          */
-        allowMethodWithConditions: function(verb, resource, conditions) {
-            addMethod.call(this, "allow", verb, resource, conditions);
-        },
+        var addMethod = function (effect, verb, resource, conditions) {
+            if (verb != "*" && !AuthPolicy.HttpVerb.hasOwnProperty(verb)) {
+                throw new Error("Invalid HTTP verb " + verb + ". Allowed verbs in AuthPolicy.HttpVerb");
+            }
+
+            if (!this.pathRegex.test(resource)) {
+                throw new Error("Invalid resource path: " + resource + ". Path should match " + this.pathRegex);
+            }
+
+            var cleanedResource = resource;
+            if (resource.substring(0, 1) == "/") {
```

Comment:
===. I am seeing this in multiple places. Not sure if you are updating this as well since i see lot of other improvements you are doing anyways.